diff --git a/internal/core/build.sh b/internal/core/build.sh
index 051a42e5f..ed26831d4 100755
--- a/internal/core/build.sh
+++ b/internal/core/build.sh
@@ -149,7 +149,6 @@ CMAKE_CMD="cmake \
 -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX}
 -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
 -DOpenBLAS_SOURCE=AUTO \
--DCMAKE_CUDA_COMPILER=${CUDA_COMPILER} \
 -DBUILD_COVERAGE=${BUILD_COVERAGE} \
 -DENABLE_CPU_PROFILING=${PROFILING} \
 -DMILVUS_GPU_VERSION=${GPU_VERSION} \
diff --git a/internal/core/thirdparty/knowhere/CMakeLists.txt b/internal/core/thirdparty/knowhere/CMakeLists.txt
index e6465b181..116ace368 100644
--- a/internal/core/thirdparty/knowhere/CMakeLists.txt
+++ b/internal/core/thirdparty/knowhere/CMakeLists.txt
@@ -12,8 +12,8 @@
 #-------------------------------------------------------------------------------

 # Update KNOWHERE_VERSION for the first occurrence
-set( KNOWHERE_VERSION v2.2.5 )
-set( GIT_REPOSITORY  "https://github.com/zilliztech/knowhere.git")
+set( KNOWHERE_VERSION 36372b1 )
+set( GIT_REPOSITORY  "https://github.com/simshi/knowhere.git")
 if ( INDEX_ENGINE STREQUAL "cardinal" )
     set( KNOWHERE_VERSION 2.2.6 )
     set( GIT_REPOSITORY  "https://github.com/zilliztech/knowhere-cloud.git")
diff --git a/internal/querynodev2/segments/cgo_util.go b/internal/querynodev2/segments/cgo_util.go
index f56c3fb2c..2e985c6c9 100644
--- a/internal/querynodev2/segments/cgo_util.go
+++ b/internal/querynodev2/segments/cgo_util.go
@@ -47,11 +47,11 @@ func HandleCStatus(ctx context.Context, status *C.CStatus, extraInfo string, fie
 	errorMsg := C.GoString(status.error_msg)
 	defer C.free(unsafe.Pointer(status.error_msg))

-	log.Ctx(ctx).With(fields...).
+	log := log.Ctx(ctx).With(fields...).
 		WithOptions(zap.AddCallerSkip(1)) // Add caller stack to show HandleCStatus caller

 	err := merr.SegcoreError(int32(errorCode), errorMsg)
-	log.Warn("CStatus returns err", zap.Error(err))
+	log.Warn("CStatus returns err", zap.Error(err), zap.String("extra", extraInfo))
 	return err
 }

diff --git a/pkg/util/indexparamcheck/conf_adapter_mgr.go b/pkg/util/indexparamcheck/conf_adapter_mgr.go
index 5099da0ca..1a4a114df 100644
--- a/pkg/util/indexparamcheck/conf_adapter_mgr.go
+++ b/pkg/util/indexparamcheck/conf_adapter_mgr.go
@@ -43,6 +43,8 @@ func (mgr *indexCheckerMgrImpl) GetChecker(indexType string) (IndexChecker, erro
 }

 func (mgr *indexCheckerMgrImpl) registerIndexChecker() {
+	mgr.checkers[IndexGpuFlat] = newFlatChecker()
+	mgr.checkers[IndexGpuIvfSQ8] = newIVFBaseChecker()
 	mgr.checkers[IndexRaftIvfFlat] = newIVFBaseChecker()
 	mgr.checkers[IndexRaftIvfPQ] = newRaftIVFPQChecker()
 	mgr.checkers[IndexFaissIDMap] = newFlatChecker()
diff --git a/pkg/util/indexparamcheck/index_type.go b/pkg/util/indexparamcheck/index_type.go
index ebef1bc7a..ba42b334f 100644
--- a/pkg/util/indexparamcheck/index_type.go
+++ b/pkg/util/indexparamcheck/index_type.go
@@ -17,6 +17,8 @@ type IndexType = string
 // IndexType definitions
 const (
 	IndexGpuBF           IndexType = "GPU_BRUTE_FORCE"
+	IndexGpuFlat         IndexType = "GPU_FLAT"
+	IndexGpuIvfSQ8       IndexType = "GPU_IVF_SQ8"
 	IndexRaftIvfFlat     IndexType = "GPU_IVF_FLAT"
 	IndexRaftIvfPQ       IndexType = "GPU_IVF_PQ"
 	IndexRaftCagra       IndexType = "GPU_CAGRA"
@@ -33,6 +35,8 @@ const (

 func IsGpuIndex(indexType IndexType) bool {
 	return indexType == IndexGpuBF ||
+		indexType == IndexGpuFlat ||
+		indexType == IndexGpuIvfSQ8 ||
 		indexType == IndexRaftIvfFlat ||
 		indexType == IndexRaftIvfPQ ||
 		indexType == IndexRaftCagra
diff --git a/scripts/core_build.sh b/scripts/core_build.sh
index bd24d116e..0b1e78250 100755
--- a/scripts/core_build.sh
+++ b/scripts/core_build.sh
@@ -236,7 +236,6 @@ ${CMAKE_EXTRA_ARGS} \
 -DBUILD_UNIT_TEST=${BUILD_UNITTEST} \
 -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX}
 -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
--DCMAKE_CUDA_COMPILER=${CUDA_COMPILER} \
 -DCMAKE_LIBRARY_ARCHITECTURE=${arch} \
 -DBUILD_COVERAGE=${BUILD_COVERAGE} \
 -DMILVUS_GPU_VERSION=${GPU_VERSION} \
